[DOC]Restructuring (#3890)
pachyderm/pachyderm: doc/concepts/data-concepts/repo.md

# Repository

A Pachyderm repository is a location where you store your data inside
Pachyderm. A Pachyderm repository is a top-level data object that contains
files and folders. Similar to Git, a Pachyderm repository tracks all
changes to the data and creates a history of data modifications that you
can access and review. You can store any type of file in a Pachyderm repo,
including binary and plain text files.

Unlike a Git repository that stores history in a `.git` file in your copy
of a Git repo, Pachyderm stores the history of your commits in a centralized

> location in the etcd key-value store. Because of that, you do not run into

Reviewer: ```suggestion
location. Because of that, you do not run into
```

While you are correct it's stored in etcd, that info is needed here. Also, that might change actually at some point and there's no reason to need this doc to change when it does.

merge conflicts as you often do with Git commits when you try to merge
your `.git` history with the master copy of the repo. With large datatsets
resolving a merge conflict might not be possible.

A Pachyderm repository is the first entity that you configure to create

> a standard pipeline. You can create a repository by running the

Reviewer: ```suggestion
A Pachyderm repository is the first entity that you configure when you want to add data to Pachyderm
```

Technically, creating the repo has nothing to do with the pipeline. They could be using pachyderm for jsut data versioning and never use pipelines.

> `pachctl create repo <name>` command or by using the Pachyderm UI. After

Reviewer: > `pachctl create repo <name>`

I'm not going to flag every command, but whatever standards we land on, obviously update it everywhere throughout these docs :)

creating the repository, you can add your data by using the

> `pachctl put file <repo@branch:/path><path-to-filesystem>` command. The

Reviewer: > `pachctl put file <repo@branch:/path><path-to-filesystem>`

This command isn't quite right. What are you going for here? You can probably simplify this by just sayinf "by using the `put file` command. 

Note: pachctl is just the command line interface. Technically, you can use the python client, the UI, the Go client, or the grpc API directly to add data to pachyderm. I probably missed this in a bunch of other places above, but we don't need to actually have copy/paste-able commands in these case. Just say, "you add data to a repo by using 'put file'" and then they can to go elsewhere (e,g. "getting you data in pachyderm" doc to get the actual command syntax for put file because otherwise this becomes a lot of superfluous information when we're really just talking about repos.

> path to the filesystem can be a local directory, file, or a URL.

Reviewer: > The path to the filesystem can be a local directory, file, or a URL.

This is covered in other places where we talk about put file. No need to have it here. 

> This repository is called the *input repo* and is stored under
`pfs/<repo-name>`. All the input data for your pipeline is stored in
this repository.

I see what you're going for here, but this isn't quite right. Instead:

"Repo can be **input repos**, meaning users or external applications outside of Pachyderm are adding data to them or they can be **output repos** which are automatically created and written to by Pachyderm pipelines only (see LINK to pipelines.md for more info on pipelines)."

This repository is called the *input repo* and is stored under
`pfs/<repo-name>`. All the input data for your pipeline is stored in
this repository.

> After you create a repository, you can specify that repository as a

Reviewer: > After you create a repository, you can specify that repository as a
PFS input in your pipeline. When the pipeline detects
new, unprocessed data in the repository, it runs your code against it.

remove this, it'll be covered in Pipelines. No need to bloat the sections. If you want, you can change this to something like, "Both outside applications and Pachyderm pipelines can read and process data from repos".

> For each pipeline, Pachyderm automatically creates a `pfs/out` directory
in the `pachd` container. This directory is called the *output repository*.
All your output data is stored in this directory.

This is incorrect. As above, we should just remove this section, it belongs in pipelines, but I'll explain what's wrong and we should discuss in TOH. This stuff is tricky -- it's not just about picking up on the names of things (e.g. pfs/out, pachd, etc), but actually building an understanding of the concepts and how they interact. That's exactly what I'm here for and would love to go over this with you as many times as you want. :D

- pfs/out is not created in the pachd container, it's created in the user container
- the above is not done for each pipeline, it's actually done for each datum within a job (to be discussed in Jobs/datums/pipelines)
- The directory is not called the output repository. The output repository is called the output repository :P. That directory is pfs/out, which is where the user code should write data that it want put into the output repository. 
- Output data is not stored in the pfs/out directory. In fact, the pfs/out directory is wiped clean after every datum. The output data is stored in the output repo.

can access and review. You can store any type of file is in a Pachyderm repo,
including binary and plain text files.

Unlike a Git repository that stores history in a `.git` file in your copy
of a Git repo, Pachyderm stores the history of your commits in a centralized
location. Because of that, you do not run into
merge conflicts as you often do with Git commits when you try to merge
your `.git` history with the master copy of the repo. With large datasets
resolving a merge conflict might not be possible.

A Pachyderm repository is the first entity that you configure when you
want to add data to Pachyderm. You can create a repository by running
the `pachctl create repo` command or by using the Pachyderm UI. After
creating the repository, you can add your data by using the
`pachctl put file` command.

The following types of repositories exist in Pachyderm:

Input repositories
 Users or external applications outside of Pachyderm can add data to
 the input repositories for further processing.

Output repositories
 Pachyderm automatically creates output repositories
 pipelines write results of computations into these repositories.

>  The output repository is the `pfs/out` located in your container.

Reviewer: This can be phrased better:

```suggestion
Any data written to the `pfs/out` directory within your pipeline user container (see pipelines LINK) will be written to that pipeline's output repository.
```

You can view the list of repositories in your Pachyderm cluster
by running the `pachctl list repo` command.

**Example:**

```bash
$ pachctl list repo
NAME     CREATED     SIZE (MASTER)
raw_data 6 hours ago 0B
```

The `pachctl inspect repo` command provides a more detailed overview
of a specified repository.

**Example:**

```bash
$ pachctl inspect repo raw_data
Name: raw_data
Description: A raw data repository
Created: 6 hours ago
Size of HEAD on master: 5.121MiB
```

If you need to delete a repository, you can run the
`pachctl delete command`. This command deletes all
data and the information about the specified
repository, such as commit history. The delete
operation is irreversible and results in a

> complete cleanup of your Pachyderm cluster.

Reviewer: ```suggestion
complete cleanup of your Pachyderm repository.
```
